File: examples/faker/person.test.js
import { describe, expect } from "https://jslib.k6.io/k6chaijs/4.3.4.3/index.js";
import { Faker } from "k6/x/faker";

export default function () {
  describe("person", (t) => {
    let faker = new Faker(11);
    expect(faker.person.firstName(), "firstName()").to.equal("Josiah");
    faker = new Faker(11);
    expect(faker.person.lastName(), "lastName()").to.equal("Abshire");
    faker = new Faker(11);
    expect(faker.person.prefix(), "prefix()").to.equal("Mr.");
    faker = new Faker(11);
    expect(faker.person.suffix(), "suffix()").to.equal("Sr.");
    faker = new Faker(11);
    expect(faker.person.jobTitle(), "jobTitle()").to.equal("Representative");
    faker = new Faker(11);
    expect(faker.person.jobDescriptor(), "jobDescriptor()").to.equal("Internal");
    faker = new Faker(11);
    expect(faker.person.jobLevel(), "jobLevel()").to.equal("Identity");
    faker = new Faker(11);
    expect(faker.person.sexType(), "sexType()").to.equal("male");
  });
}

File: examples/faker/person_test.go
package faker

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func person(t *testing.T) goPerson {
	t.Helper()

	return &goPersonImpl{fakeit: fakeit(t)}
}

func Test_goPerson(t *testing.T) {
	t.Parallel()
	must := noError(t)

	assert.Equal(t, "Josiah", must(person(t).firstNameMethod()))
	assert.Equal(t, "Abshire", must(person(t).lastNameMethod()))
	assert.Equal(t, "Mr.", must(person(t).prefixMethod()))
	assert.Equal(t, "Sr.", must(person(t).suffixMethod()))
	assert.Equal(t, "Representative", must(person(t).jobTitleMethod()))
	assert.Equal(t, "Internal", must(person(t).jobDescriptorMethod()))
	assert.Equal(t, "Identity", must(person(t).jobLevelMethod()))
	assert.Equal(t, "male", must(person(t).sexTypeMethod()))
}
